### PR TITLE
You can now fasten the TEG circulators

### PR DESCRIPTION
**Pull request by Crazylemon64**
:cl:Crazylemon
rscadd: You can now fasten the circulators in the TEG kit with a wrench - no adminbus needed to set it up, now
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
Wait, seriously? You could never set this up yourself before?

**Comment by LiquidNightfury** [comment]
gibe more engine pls

**Comment by Crazylemon64** [comment]
Don't think you could - the circulators couldn't be fastened without adminry

**Comment by LiquidNightfury** [comment]
Can confirm this didn't work before. Time for a combination turbine/TEG anyone?

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 1]
```diff
@@ -66,3 +66,13 @@
...
 	return 1
+
+/obj/machinery/atmospherics/binary/circulator/attackby(var/obj/item/W as obj, var/mob/user as mob, params)
+	if(istype(W, /obj/item/weapon/wrench))
+		var/turf/T = get_turf(src)
+		if(!istype(T, /turf/simulated))
+			return ..()
+		anchored = !anchored
+		to_chat(user, "You [(anchored) ? "fasten" : "loosen"] \the [src] to the floor")
+		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
```
Comment (PJB3005) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: Src is always implied.

**Comment by PJB3005** [comment]
TEG turbine sounds like a terrible idea.

**Comment by Bigtimetripod** [comment]
Can confirm this did not work, i usually surronded the exchangers in glass so people and me would nof pump into em. Also hopefully this fixes the problem that TEG has to be horizontal for it to work, when it was vertical, gases woult not pass thru the exchangers.

**Comment by Crazylemon64** [comment]
@Bigtimetripod I did not add any changes to make the circulators work when turned - that will need to come later

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 2]
```diff
@@ -66,3 +66,13 @@
 		icon_state = "circ-off"
 
 	return 1
+
+/obj/machinery/atmospherics/binary/circulator/attackby(var/obj/item/W as obj, var/mob/user as mob, params)
+	if(istype(W, /obj/item/weapon/wrench))
+		var/turf/T = get_turf(src)
+		if(!istype(T, /turf/simulated))
+			return ..()
+		anchored = !anchored
```
Comment (PJB3005) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: Gonna need more than that to fasten something to a pipe network, no?

Comment (Crazylemon64) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: It generally behaves like an endpoint for most pipes as-is - but our atmos machinery code is a bit of a mess, yes.

**Pull request merged**